packages/worker/src/worker.ts

```diff
@@ -1133,13 +1137,16 @@ export class Worker {
                       if (
                         !(
                           startWorkflow &&
-                          startWorkflow.workflowId &&
-                          startWorkflow.workflowType &&
-                          startWorkflow.randomnessSeed
+                          startWorkflow.workflowId != null &&
+                          startWorkflow.workflowType != null &&
+                          startWorkflow.randomnessSeed != null &&
+                          startWorkflow.firstExecutionRunId != null &&
+                          startWorkflow.attempt != null &&
+                          startWorkflow.startTime != null
                         )
                       ) {
                         throw new TypeError(
-                          `Expected StartWorkflow with workflowId, workflowType and randomnessSeed, got ${JSON.stringify(
+                          `Expected StartWorkflow with workflowId, workflowType, randomnessSeed, firstExecutionRunId, attempt and startTime. Got ${JSON.stringify(
                             maybeStartWorkflow
                           )}`
                         );
```

Comment:
nit: At this point just say:

```suggestion
                        throw new TypeError(`Malformed StartWorkflow: ${JSON.stringify(maybeStartWorkflow)}`);
```